tests/bluetooth/bsim_bt/bsim_test_audio/src/bass_client_test.c

```diff
@@ -0,0 +1,536 @@
+	sync_create_param.timeout = 0xa;
+	err = bt_le_per_adv_sync_create(&sync_create_param, &g_pa_sync);
+	if (err != 0) {
+		FAIL("Could not create PA syncs (err %d)\n", err);
+		return;
+	}
+	WAIT_FOR(g_pa_synced);
+	printk("PA synced\n");
+}
+
+static void test_bass_add_source(void)
+{
+	int err;
+	struct bt_bass_add_src_param add_src_param = { 0 };
+	struct bt_bass_subgroup subgroup = { 0 };
+
+	printk("Adding source\n");
+	g_cb = g_write_complete = false;
+	bt_addr_le_copy(&add_src_param.addr, &g_broadcaster_addr);
+	add_src_param.adv_sid = g_broadcaster_info.sid;
+	add_src_param.num_subgroups = 1;
+	add_src_param.pa_interval = g_broadcaster_info.interval;
+	add_src_param.pa_sync = false;
+	add_src_param.broadcast_id = g_broadcast_id;
+	add_src_param.subgroups = &subgroup;
+	subgroup.bis_sync = 0;
+	subgroup.metadata_len = 0;
+	err = bt_bass_client_add_src(g_conn, &add_src_param);
+	if (err != 0) {
+		FAIL("Could not add source (err %d)\n", err);
+		return;
+	}
+	WAIT_FOR(g_cb && g_write_complete);
+	printk("Source added\n");
+}
+
+static void test_bass_mod_source(void)
+{
+	int err;
+	struct bt_bass_mod_src_param mod_src_param = { 0 };
+	struct bt_bass_subgroup subgroup = { 0 };
+
+	printk("Modify source\n");
+	g_cb = g_write_complete = false;
+	mod_src_param.src_id = g_src_id;
+	mod_src_param.num_subgroups = 1;
+	mod_src_param.pa_sync = true;
+	mod_src_param.subgroups = &subgroup;
+	mod_src_param.pa_interval = g_broadcaster_info.interval;
+	subgroup.bis_sync = 0;
+	subgroup.metadata_len = 0;
+	err = bt_bass_client_mod_src(g_conn, &mod_src_param);
+	if (err != 0) {
+		FAIL("Could not modify source (err %d)\n", err);
+		return;
+	}
+	WAIT_FOR(g_cb && g_write_complete);
+	printk("Source added, waiting for server to PA sync\n");
+	WAIT_FOR(g_state_synced)
+	printk("Server PA synced\n");
+}
+
+static void test_bass_broadcast_code(void)
+{
+	uint8_t broadcast_code[BT_BASS_BROADCAST_CODE_SIZE];
+	int err;
+
+	for (int i = 0; i < ARRAY_SIZE(broadcast_code); i++) {
+		broadcast_code[i] = i;
+	}
+
+	printk("Adding broadcast code\n");
+	g_write_complete = false;
+	err = bt_bass_client_broadcast_code(g_conn, g_src_id, broadcast_code);
+	if (err != 0) {
+		FAIL("Could not add broadcast code (err %d)\n", err);
+		return;
+	}
+	WAIT_FOR(g_write_complete);
+	printk("Broadcast code added\n");
+}
+
+static void test_bass_remove_source(void)
+{
+	int err;
+
+	printk("Removing source\n");
+	g_cb = g_write_complete = false;
+	err = bt_bass_client_rem_src(g_conn, g_src_id);
+	if (err != 0) {
+		FAIL("Could not remove source (err %d)\n", err);
+		return;
+	}
+	WAIT_FOR(g_cb && g_write_complete);
+	printk("Source removed\n");
+}
+
+static void test_main(void)
+{
+	int err;
+
+	err = bt_enable(NULL);
+
+	if (err != 0) {
+		FAIL("Bluetooth enable failed (err %d)\n", err);
+		return;
+	}
+
+	bt_conn_cb_register(&conn_callbacks);
+	bt_bass_client_register_cb(&bass_cbs);
+	printk("Periodic Advertising callbacks register...");
```

Comment:
nitpick: Missing newline character at the end of printk message?

Reply:
Removed; we don't really need this one.